include/sparrow/layout/variable_size_binary_view_array.hpp

```diff
@@ -0,0 +1,369 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstddef>
+#include <ranges>
+
+#include "sparrow/arrow_interface/arrow_array.hpp"
+#include "sparrow/arrow_interface/arrow_schema.hpp"
+#include "sparrow/utils/functor_index_iterator.hpp"
+#include "sparrow/arrow_interface/arrow_schema.hpp"
+#include "sparrow/arrow_array_schema_proxy.hpp"
+#include "sparrow/layout/array_bitmap_base.hpp"
+#include "sparrow/layout/layout_utils.hpp"
+#include "sparrow/utils/iterator.hpp"
+#include "sparrow/utils/nullable.hpp"
+#include "sparrow/layout/array_access.hpp"
+#include "sparrow/buffer/dynamic_bitset.hpp"
+#include "sparrow/buffer/u8_buffer.hpp"
+#include "sparrow/utils/ranges.hpp"
+
+namespace sparrow
+{
+
+
+    template <class T>
+    class variable_size_binary_view_array_impl;
+    using string_view_array = variable_size_binary_view_array_impl<std::string_view>;
+    using binary_view_array = variable_size_binary_view_array_impl<std::span<const std::byte>>;
```

Comment:
Since `std::span` is a const view, I don't think it is necessary to constify its value type. I would prefer that we use buffer_view instead, since `std::span` implementation is very incomplete (no comparison operators, no cbegin/cend methods, etc...), but I struggle to get it working with the variable_size_binary_array.